test/tools/Modules/HttpsListener/HttpsListener.psm1

```diff
@@ -0,0 +1,126 @@
+Class HttpsListener 
+{
+    [int]$Port
+    [System.Management.Automation.Job]$Job
+    
+    HttpsListener () { }
+
+    [string] ToString() {
+        return ('Port: {0}; Status: {1}' -f $This.Port, $This.JobStateInfo.State)
+    }
+}
+
+$HttpsListeners = @{}
+
+function Get-HttpsListener 
+{
+    [CmdletBinding(ConfirmImpact = 'Low')]
+    [OutputType([HttpsListener])]
+    param 
+    (
+        [ValidateRange(1,65535)]
+        [int[]]$Port
+    )
+
+    process 
+    {
+        if ($Port)
+        {
+            return $Script:HttpsListeners[$port]
```

Comment:
We only need to do this if we run tests in parallel. I think we're never going to do this (no Pester support is and CI VMs is very slow). So to cache config and to reinitialize the listener if the config changed is enough.